docs: update terminology in training material

jf205 opened the pull request:
In this PR I've reviewed the use of QL in the training presentations. The changes include:
- changing QL -> CodeQL where we are talking about code analysis
- deleting QL in places where context is clear
- change snapshot to database or CodeQL database as appropriate

I've also included a couple of changes that were made to the 1.22 branch---I thought they would have made their way into master 🤷‍♂ 

@lcartey, please could you check that you are happy with these changes?
@shati-patel and @felicitymay, could you review too please?

Previews: 
http://docteam.internal.semmle.com/james/ql-training/terminology/cpp/
http://docteam.internal.semmle.com/james/ql-training/terminology/java/

shati-patel commented on docs/language/learn-ql/ql-training.rst:
```diff
@@ -1,19 +1,19 @@
-The QL language is easy to learn, and exploring code using QL is the most efficient way to perform variant analysis. 
+CodeQL is easy to learn, and exploring code using CodeQL is the most efficient way to perform variant analysis. 
 
 Learning QL for variant analysis
```
```suggestion
Learning CodeQL for variant analysis
```

jf205 replied:
It seems I didn't push the commit that updated these titles 🤦‍♂ 
Sorry for creating the extra reviewing work!

shati-patel commented on docs/language/learn-ql/ql-training.rst:
```diff
@@ -1,19 +1,19 @@
 
 Learning QL for variant analysis
-================================
+--------------------------------
```
```suggestion
------------------------------------
```

shati-patel commented on docs/language/learn-ql/ql-training.rst:
```diff
@@ -1,19 +1,19 @@
+CodeQL training and variant analysis examples
+=============================================
 
 QL and variant analysis
```
```suggestion
CodeQL and variant analysis
```

shati-patel commented on docs/language/learn-ql/ql-training.rst:
```diff
@@ -1,19 +1,19 @@
 
 QL and variant analysis
-=======================
+-----------------------
```
```suggestion
---------------------------
```

shati-patel commented on docs/language/learn-ql/ql-training.rst:
```diff
@@ -24,43 +24,43 @@ Start learning how to use QL in variant analysis for a specific language by look
-- `Introduction to global data flow <../ql-training/cpp/global-data-flow-cpp.html>`__–an introduction to analyzing global data flow in C/C++ using QL.
-- `Analyzing control flow: QL for C/C++  <../ql-training/cpp/control-flow-cpp.html>`__–an introduction to analyzing control flow in C/C++ using QL.
+- `Introduction to global data flow <../ql-training/cpp/global-data-flow-cpp.html>`__–an introduction to analyzing global data flow in C/C++ using CodeQL.
+- `Analyzing control flow: CodeQL for C/C++  <../ql-training/cpp/control-flow-cpp.html>`__–an introduction to analyzing control flow in C/C++ using QL.
```
```suggestion
- `Analyzing control flow: CodeQL for C/C++  <../ql-training/cpp/control-flow-cpp.html>`__–an introduction to analyzing control flow in C/C++ using CodeQL.
```

shati-patel commented on docs/language/learn-ql/ql-training.rst:
```diff
@@ -24,43 +24,43 @@ Start learning how to use QL in variant analysis for a specific language by look
-- To find more detailed information about how to write QL queries for specific languages, visit the links in :ref:`Writing QL queries <writing-ql-queries>`.
-- To read more about how QL queries have been used in Semmle's security research, and to read about new QL developments, visit the `Semmle blog <https://blog.semmle.com>`__. 
+- If you are completely new to CodeQL, look at our introductory topics in :doc:`Learning CodeQL <index>`.
+- To find more detailed information about how to write CodeQL queries for specific languages, visit the links in :ref:`Writing CodeQL queries <writing-ql-queries>`.
```
```suggestion
- To find more detailed information about how to write queries for specific languages, visit the links in :ref:`Writing CodeQL queries <writing-ql-queries>`.
```

(Perhaps remove this to slightly minimise the number of "CodeQL"s.)

shati-patel commented on docs/language/ql-training/cpp/data-flow-cpp.rst:
```diff
@@ -86,9 +82,9 @@ Write a query that flags ``printf`` calls where the format argument is not a ``S
 .. note::
 
-  This first query is about finding places where the format specifier is not a constant string. In QL for C/C++, constant strings are modeled as ``StringLiteral`` nodes, so we are looking for calls to format functions where the format specifier argument is not a string literal.
+  This first query is about finding places where the format specifier is not a constant string. In CodeQL for C/C++, constant strings are modeled as ``StringLiteral`` nodes, so we are looking for calls to format functions where the format specifier argument is not a string literal.
```
```suggestion
  This first query is about finding places where the format specifier is not a constant string. In the CodeQL libraries for C/C++, constant strings are modeled as ``StringLiteral`` nodes, so we are looking for calls to format functions where the format specifier argument is not a string literal.
```

(To avoid "in CodeQL ...")

shati-patel commented on docs/language/ql-training/cpp/data-flow-cpp.rst:
```diff
@@ -86,9 +82,9 @@ Write a query that flags ``printf`` calls where the format argument is not a ``S
+  This first query is about finding places where the format specifier is not a constant string. In CodeQL for C/C++, constant strings are modeled as ``StringLiteral`` nodes, so we are looking for calls to format functions where the format specifier argument is not a string literal.
 
-  The `C/C++ standard libraries <https://help.semmle.com/qldoc/cpp/>`__ include many different formatting functions that may be vulnerable to this particular attack–including ``printf``, ``snprintf``, and others. Furthermore, each of these different formatting functions may include the format string in a different position in the argument list. Instead of laboriously listing all these different v ...
+  The `C/C++ standard libraries <https://help.semmle.com/qldoc/cpp/>`__ include many different formatting functions that may be vulnerable to this particular attack–including ``printf``, ``snprintf``, and others. Furthermore, each of these different formatting functions may include the format string in a different position in the argument list. Instead of laboriously listing all these different v ...
```
```suggestion
  The `C/C++ standard libraries <https://help.semmle.com/qldoc/cpp/>`__ include many different formatting functions that may be vulnerable to this particular attack–including ``printf``, ``snprintf``, and others. Furthermore, each of these different formatting functions may include the format string in a different position in the argument list. Instead of laboriously listing all these different variants, we can make use of the standard CodeQL class ``FormattingFunction``, which provides an interface that models common formatting functions in C/C++.
```

shati-patel commented on docs/language/ql-training/cpp/intro-ql-cpp.rst:
```diff
@@ -110,14 +106,14 @@ Each query library also implicitly defines a module.
+  Queries are always contained in query files with the file extension ``.ql``. `Quick queries <https://help.semmle.com/ql-for-eclipse/Content/WebHelp/quick-query.html>`__, run in `QL for Eclipse <https://help.semmle.com/ql-for-eclipse/Content/WebHelp/home-page.html>`__, are no exception: the quick query window maintains a temporary query file in the background.
 
-  Parts of queries can be lifted into `QL library files <https://help.semmle.com/QL/ql-handbook/modules.html#library-modules>`__ with the extension ``qll``. Definitions within such libraries can be brought into scope using ``import`` statements, and similarly QLL files can import each other’s definitions using “import” statements.
+  Parts of queries can be lifted into `library files <https://help.semmle.com/QL/ql-handbook/modules.html#library-modules>`__ with the extension ``qll``. Definitions within such libraries can be brought into scope using ``import`` statements, and similarly QLL files can import each other’s definitions using “import” statements.
```
```suggestion
  Parts of queries can be lifted into `library files <https://help.semmle.com/QL/ql-handbook/modules.html#library-modules>`__ with the extension ``.qll``. Definitions within such libraries can be brought into scope using ``import`` statements, and similarly QLL files can import each other’s definitions using “import” statements.
```

shati-patel commented on docs/language/ql-training/cpp/program-representation-cpp.rst:
```diff
@@ -66,9 +62,9 @@ Working with variables
+Functions are represented by the Function class. Each declaration or definition of a function is represented by a ``FunctionDeclarationEntry``.
 
-Calls to functions are modeled by QL class Call and its subclasses:
+Calls to functions are modeled by class Call and its subclasses:
```
```suggestion
Calls to functions are modeled by class ``Call`` and its subclasses:
```

shati-patel commented on docs/language/ql-training/cpp/program-representation-cpp.rst:
```diff
@@ -107,7 +103,7 @@ Working with macros
     y = square(y0), z = square(z0)
 
-is represented in the snapshot database as:
+is represented in the CodeQL database database as:
```
```suggestion
is represented in the CodeQL database as:
```

shati-patel commented on docs/language/ql-training/java/intro-ql-java.rst:
```diff
@@ -162,7 +158,7 @@ Member predicates are inherited and can be overridden.
   In the example, declaring a variable “EmptyBlock e” will allow it to range over only those blocks that have zero statements.
 
-Classes in QL continued
+Classes in continued
```
```suggestion
Classes continued
```

shati-patel commented on docs/language/ql-training/slide-snippets/intro-ql-general.rst:
```diff
@@ -105,33 +105,33 @@ Analysis overview
 - a **declarative** language without side effects
 - an **object-oriented** language
-- a **query** language working on a read-only snapshot database
+- a **query** language working on a read-only CodeQL database database
```
```suggestion
- a **query** language working on a read-only CodeQL database
```

jf205 commented:
Thanks for the reviews. It was a bit of a rush---sorry for letting things slip through 😞

shati-patel commented on docs/language/ql-training/slide-snippets/local-data-flow.rst:
```diff
@@ -61,16 +61,16 @@ Data flow graphs
   For further information, see:
 
-  - `Introduction to data flow analysis in QL <https://help.semmle.com/QL/learn-ql/ql/intro-to-data-flow.html>`__
+  - `Introduction to data flow analysis in CodeQL <https://help.semmle.com/QL/learn-ql/ql/intro-to-data-flow.html>`__
```
```suggestion
  - `Introduction to data flow analysis with CodeQL <https://help.semmle.com/QL/learn-ql/ql/intro-to-data-flow.html>`__
```

shati-patel commented on docs/language/ql-training/cpp/control-flow-cpp.rst:
```diff
@@ -2,11 +2,7 @@
 Analyzing control flow
```
Can't comment inline, but [line 92](https://github.com/Semmle/ql/pull/2257/files#diff-7c46c31059ce63c74c0f2febd77649b3R92) of this file should talk about the "(CodeQL) class" instead of "QL class".

shati-patel commented on docs/language/ql-training/java/program-representation-java.rst:
```diff
@@ -2,11 +2,7 @@
 Program representation
```
[Line 26](https://github.com/Semmle/ql/pull/2257/files#diff-e4161e72680fe0e0cf7d21153fd0a09fR26) of this file also mentions "QL class".

shati-patel replied:
Similarly, replace QL with CodeQL on [line 50](https://github.com/Semmle/ql/pull/2257/files#diff-e4161e72680fe0e0cf7d21153fd0a09fR50).

shati-patel replied:
... and remove QL (or replace with CodeQL) on lines [77](https://github.com/Semmle/ql/pull/2257/files#diff-e4161e72680fe0e0cf7d21153fd0a09fR77) and [79](https://github.com/Semmle/ql/pull/2257/files#diff-e4161e72680fe0e0cf7d21153fd0a09fR79).  (I _think_ that's all for this file!)

jf205 replied:
@lcartey: line 50 says: _QL also classifies `void` and `<nulltype>` (the type of the `null` literal) as primitive types._

Are these primitive types specific to Java analysis?

lcartey commented:
At some point we will need to update the links to QL for Eclipse, and replace them with the VSCode CodeQL extension. This will need to be post-universe (as it will only become publicly available then).

shati-patel commented on docs/language/ql-training/slide-snippets/abstract-syntax-tree.rst:
```diff
@@ -47,23 +47,23 @@ The basic representation of an analyzed program is an *abstract syntax tree (AST
 
 Each entity belongs to an entity type. Entity types have names starting with “@” and are defined in the database schema (not in QL).
 
 Properties of AST nodes and their relationships to each other are encoded by database relations, which are predicates defined in the database (not in QL).
```
(I'm not sure about the usage of QL here, but it's probably not worth worrying about for now!)

jf205 replied:
Yeah, I left this as-is because I couldn't see anything specific to code 🤷‍♂

lcartey commented on docs/language/ql-training/cpp/program-representation-cpp.rst:
```diff
@@ -25,8 +21,8 @@ Agenda
 
-AST QL classes
-==============
+AST classes
```
This is one file where I would retain the "QL" - otherwise new query authors can get confused with classes in the target language.

jf205 replied:
`AST CodeQL classes` as they are specific to C++ analysis?

lcartey commented on docs/language/ql-training/cpp/program-representation-cpp.rst:
```diff
@@ -66,9 +62,9 @@ Working with variables
+Functions are represented by the Function class. Each declaration or definition of a function is represented by a ``FunctionDeclarationEntry``.
 
-Calls to functions are modeled by QL class Call and its subclasses:
+Calls to functions are modeled by class ``Call`` and its subclasses:
```
```suggestion
Calls to functions are modeled by the QL class ``Call`` and its subclasses:
```

jf205 replied:
CodeQL class, I think, for the same reason as above.

lcartey commented on docs/language/ql-training/java/intro-ql-java.rst:
```diff
@@ -72,9 +68,9 @@ A simple QL query
+   A `query <https://help.semmle.com/QL/ql-handbook/queries.html>`__ consists of a “select” clause that indicates what results should be returned. Typically it will also provide a “from” clause to declare some variables, and a “where” clause to state conditions over those variables. For more information on the structure of query files (including links to useful topics in the `QL language handbook ...
 
-   In our example here, the first line of the query imports the `Java standard QL library <https://help.semmle.com/qldoc/java/>`__, which defines concepts like ``IfStmt`` and ``Block``.
+   In our example here, the first line of the query imports the `CodeQL for Java library <https://help.semmle.com/qldoc/java/>`__, which defines concepts like ``IfStmt`` and ``Block``.
```
According to the terminology doc, this should be "CodeQL library for Java"
https://docs.google.com/document/d/1o8WFKQ8AN7zYq6zTvrSxsOmngRmXGE8KDTs-02KW8jM/edit

lcartey commented on docs/language/ql-training/java/intro-ql-java.rst:
```diff
@@ -134,15 +130,15 @@ A predicate allows you to pull out and name parts of a query.
+=======
 
-A QL class allows you to name a set of values and define (member) predicates on them.
+A class allows you to name a set of values and define (member) predicates on them.
```
Again, I think it's worth retaining the QL here, if we can.

jf205 replied:
Yep. It should probably be CodeQL class too.

jf205 replied:
Hmm, actually I'll leave as QL class for now as there is no mention of any code here.

lcartey commented on docs/language/ql-training/java/query-injection-java.rst:
```diff
@@ -88,10 +84,10 @@ Let’s start by looking for calls to methods with names of the form ``sparql*Qu
     - a ``Method``: the SPARQL query method.
   
-  - The ``where`` part of the query ties these three QL variables   together using `predicates <https://help.semmle.com/QL/  ql-handbook/predicates.html>`__ defined in the `standard QL for   Java library <https://help.semmle.com/qldoc/java/>`__.
+  - The ``where`` part of the query ties these variables together using `predicates <https://help.semmle.com/QL/ql-handbook/predicates.html>`__ defined in the `standard CodeQL for   Java library <https://help.semmle.com/qldoc/java/>`__.
```
CodeQL library for Java

shati-patel commented on docs/language/ql-training/cpp/program-representation-cpp.rst:
```diff
@@ -66,9 +62,9 @@ Working with variables
+Functions are represented by the Function class. Each declaration or definition of a function is represented by a ``FunctionDeclarationEntry``.
 
-Calls to functions are modeled by QL class Call and its subclasses:
+Calls to functions are modeled by CodeQL class ``Call`` and its subclasses:
```
```suggestion
Calls to functions are modeled by the CodeQL class ``Call`` and its subclasses:
```

jf205 replied:
🙇

jf205 commented:
Thanks for the suggestions. I think I've addressed all the comments. It may become apparent that clarification is needed in a few more places as others start using the slides, so I'll likely need to follow this up later.

felicitymay commented:
It sounds as if this is ready to merge.
